website/source/guides/tls.html.md

```diff
@@ -0,0 +1,375 @@
+---
+layout: "guides"
+page_title: "Securing Nomad with TLS"
+sidebar_current: "guides-tls"
+description: |-
+  Securing Nomad's cluster communication is not only important for security but
+  can even ease operations by preventing mistakes and misconfigurations. Nomad
+  optionally uses mutual TLS (mTLS) for all HTTP and RPC communication.
+---
+
+# Securing Nomad with TLS
+
+Securing Nomad's cluster communication is not only important for security but
+can even ease operations by preventing mistakes and misconfigurations. Nomad
+optionally uses mutual TLS (mTLS) for all HTTP and RPC communication. Nomad's
+use of mTLS provides the following properties:
+
+* Prevent unauthorized Nomad access
+* Prevent observing or tampering with Nomad communication
+* Prevent client/server role or region misconfigurations
+
+The 3rd property is fairly unique to Nomad's use of TLS. While most uses of TLS
+verify the identity of the server you're connecting to based on a domain name
+such as `nomadproject.io`, Nomad verifies the node you're connecting to is in
+the expected region and configured for the expected role (e.g.
+`client.us-west.nomad`).
+
+Configuring TLS can be unfortunately complex process, but if you used the
+[Getting Started Guide's Vagrantfile][Vagrantfile] or have [cfssl][] and Nomad
```

Comment:
tl;dr - Great question. I'm open to suggestions.

My goals in this guide were to keep the number of tools/services involved to an absolute minimum. The initial guide used OpenSSL instead of cfssl. However I dropped OpenSSL in favor of cfssl because OpenSSL required a combination of a large config file and confusing commands. OpenSSL also has multiple subtlety different ways of accomplishing the same task (eg the x509 vs ca subcommands) which made it hard for me to know if I was teaching best practices or not.

Using a public CA like Let's Encrypt isn't an option for reasons covered later in the guide.

Using Vault's PKI backed would be ideal, but I didn't for two reasons:

1. Lots more setup than using a simple command line tool like `cfssl`. We could do it for users in our Vagrantfile but that only further obscures the steps they'll need to take in production.
2. Gives the _impression_ of dependencies between HashiCorp products that doesn't exist. For the initial guide I want to make sure users understand they can use whatever tools they prefer.

That being said I think promoting our Vault Integration docs to a Guide that covers certificate management would be fantastic. (Even more fantastic would be be writing code to automate certificate management in Nomad clusters with Vault, but I think we're a long way from that.)